controllers/storagecluster/reconcile.go

```diff
@@ -314,7 +321,7 @@ func (r *StorageClusterReconciler) reconcilePhases(
 	}
 
 	// Add conditions if there are none
-	if instance.Status.Conditions == nil {
+	if len(instance.Status.Conditions) == 1 && instance.Status.Conditions[0].Type == ocsv1.ConditionVersionMismatch {
```

Comment:
Won't condition be nil at the beginning?

Reply:
Not really, because we are adding the version mismatch condition before this code executes while checking the spec.

Reply:
@umangachapagain do you think we should go ahead and move it before the `validateClusterSpec` function call or are we good here?

Reply:
This is fine.